meetings/__init__.py

```diff
@@ -67,18 +75,37 @@ async def log_tag(self, tag, evt: MessageEvent) -> None:
             UPDATE meeting_logs SET tag = $3 WHERE meeting_id = $1 AND timestamp = $2
           """
     await self.database.execute(dbq, meeting, timestamp, tag)
+  
+  async def change_topic(self, topic, evt: MessageEvent) -> None:
+    dbq = """
+            UPDATE meetings SET topic = $3 WHERE meeting_id = $1 AND room_id = $2
+          """
+    await self.database.execute(dbq, self.meeting_id(evt.room_id), evt.room_id, topic)
 
-  # Helper: upload a JSON file
-  async def upload_file(self, evt, items, filename):
-    if len(items) == 0:
-      return
-    
-    logs = tuple()
-    for row in items:
-      log = f"{row[1]} <{row[2]}> #{row[3]}"
-      logs += (log,)
-    
-    data = "\n".join(logs).encode("utf-8")
+    # change the topic of the ^topic message to the topic
+    timestamp = evt.timestamp
+    dbq = """
+            UPDATE meeting_logs SET topic = $3 WHERE meeting_id = $1 AND timestamp = $2
+          """
+    await self.database.execute(dbq, self.meeting_id(evt.room_id), timestamp, topic)
+
+  async def log_to_db(self, meeting, timestamp, sender, message, topic):
+      # Log the item to the db
+      dbq = """
+              INSERT INTO meeting_logs (meeting_id, timestamp, sender, message, topic) VALUES ($1, $2, $3, $4, $5)
+            """
+      await self.database.execute(dbq, meeting, timestamp, sender, message, topic)
```

Comment:
In the previous version I was logging the display_name via `self.client.get_displayname(evt.sender)`. I can see you changed that back to the plain MXID here, is there a particular reason for that?

Reply:
ah, yeah i did this because i figured the displayname is something that could be changed (even during the course of a meeting, which could also impact lines said summary etc) i figured if we want the display name later when doing the logs, we can get it from the client then?